### DISCUSSION TITLE
Error when starting Patrol example test from IDE: Binding is already initialized

### DISCUSSION THREAD

**Discussion by pandoras-toolbox** [leancodepl/patrol#1528] [Q&A]
When starting example_test.dart in the Patrol source code from Visual Studio as a test it raises an error:

> Failed to load "/home/xxx/VsCodeProjects/patrol/packages/patrol/example/integration_test/example_test.dart": 'package:flutter/src/foundation/binding.dart': Failed assertion: line 150 pos 12: '_debugInitializedType == null': Binding is already initialized to IntegrationTestWidgetsFlutterBinding
> dart:core 

Why? How is this test supposed to be started from an IDE?

Because when I start it with the command-line it works:

`patrol test --target integration_test/example_test.dart`

**Comment by bartekpacia** [comment]
Starting from IDE is not supported right now.

We do have some plans to add it in the future - see e.g #1342 and #1504.

**Reply by pandoras-toolbox** [reply]
Oh. But there is another a way to debug, to stop at a breakpoint in a test?

I think yesterday I could start it from IDE by adding a launch configuration in Visual Code and start the test file with "Run > Start Debugging" instead of starting it directly as a test. But then later in the day it did not work anymore, I could not figure out why, maybe I changed something related to it.

**Reply by bartekpacia** [reply]
For debugging, we have the [`patrol develop`](https://patrol.leancode.co/cli-commands/develop) command. Breakpoints from within IDE are not supported, but hey, you can use good old `print`s.

Running from VSCode with debugging _might_ be possible if you `flutter attach`, but I haven't tried it and we don't support it now.

**Reply by pandoras-toolbox** [reply]
Okay, thank you. I will try to use it this way for debugging purpose.

Maybe it is possible to throw a more specific error message when starting a Patrol test from the IDE, like "NotSupportedException: Starting Patrol is only supported with CLI tools"

**Reply by bartekpacia** [reply]
That's a nice idea, but there's just too much other more important stuff to do, so minor stuff like this is being pushed back. If you'd like to contribute a PR, I'm happy to review & merge it.

**Reply by pandoras-toolbox** [reply]
I understand. My Dart/Flutter knowledge is still too small for this, since until now I only used Java, where I would had that knowledge already. I am glad to contribute that when I more Dart/Flutter and Patrol skills.

I am doing a proof of concept for automating tests for our app with Patrol, it looks good to me so far, but a bit rough in the beginning.
